ts/src/main/scala/ch/epfl/ts/data/StrategyParameters.scala

```diff
@@ -144,33 +182,98 @@ object CoefficientParameter extends ParameterTrait[Double] {
       n <- 0 to (1 / resolution).toInt
     } yield (n * resolution)
   }
-  
+
   def defaultValue = 1.0
 }
 
 
 
 /**
+ * Parameter representing an integer number greater or equal to zero.
+ */
+case class NaturalNumberParameter(natural: Int) extends Parameter("NaturalNumber") {
+  type T = Int
+  def companion = NaturalNumberParameter
+  def value(): Int = natural
+}
+
+object NaturalNumberParameter extends ParameterTrait {
+  type T = Int
+  def getInstance(v: Int) = new NaturalNumberParameter(v)
+
+  /**
+   * Coefficient must lie in { 0, 1, ... }
+   */
+  def isValid(v: Int): Boolean = (v >= 0)
+
+  def validValues: Iterable[Int] = Stream.from(0)
+
+  def defaultValue = 0
+}
+
+
+
+/**
+ * Parameter representing a duration.
+ *
+ * @param duration
+ * @param unit Defaults to milliseconds
+ */
+case class TimeParameter(duration: Long, unit: TimeUnit) extends Parameter("Time") {
+  type T = FiniteDuration
+  def this(duration: Long) = this(duration, MillisecondsUnit)
+	def this(duration: FiniteDuration) = this(duration.length, duration.unit)
+
+  def companion = TimeParameter
+  def value(): FiniteDuration = FiniteDuration(duration, unit)
+}
+
+object TimeParameter extends ParameterTrait {
+  import scala.language.postfixOps
+
+  type T = FiniteDuration
+
+  def getInstance(v: Long) = new TimeParameter(v)
+  def getInstance(v: Long, u: TimeUnit) = new TimeParameter(v, u)
+  def getInstance(d: FiniteDuration) = new TimeParameter(d)
+
+  /**
+   * Duration must be positive or null
+   */
+  def isValid(v: Long): Boolean = (v >= 0)
+  def isValid(d: FiniteDuration): Boolean = (d.length >= 0)
+
+  // TODO: user-selected resolution
+  def validValues: Iterable[FiniteDuration] =
+    Stream.from(0) map { n => (100L * n) milliseconds }
+
+  def defaultValue = (0L milliseconds)
+}
+
+
+
+/**
  * Parameter representing a pair of currencies to be traded.
  */
-case class CurrencyPairParameter(currencies: (Currency, Currency)) extends Parameter[(Currency, Currency)]("CurrencyPair") {  
+case class CurrencyPairParameter(currencies: (Currency, Currency)) extends Parameter("CurrencyPair") {
+  type T = (Currency, Currency)
 	def companion = CurrencyPairParameter
 	def value(): (Currency, Currency) = currencies
 }
 
-object CurrencyPairParameter extends ParameterTrait[(Currency, Currency)] {
-	private type CurrencyPair = Tuple2[Currency, Currency]
-	def getInstance(v: CurrencyPair) = new CurrencyPairParameter(v)
-  
+object CurrencyPairParameter extends ParameterTrait {
+	type T = (Currency, Currency)
+	def getInstance(v: T) = new CurrencyPairParameter(v)
+
 	/**
 	 * All currency pairs are acceptable as long as they're not twice the same.
 	 * @TODO: It would be great if we could enforce an order inside the pair, so that we avoid having to handle swapped currency pairs
```

Comment:
As I wrote before, in my opinion this is fine and can be desirable sometimes actually.